datafusion/physical-plan/src/streaming.rs

```diff
@@ -35,6 +35,10 @@ use futures::stream::StreamExt;
 use log::debug;
 
 /// A partition that can be converted into a [`SendableRecordBatchStream`]
+///
+/// Combined with [`StreamingTable`], you can use this trait to implement
```

Comment:
I believe this is the core usecase -- can you confirm @tustvold ?

Reply:
Yup, it avoids a lot of boilerplate for the simple use-case